src/Symfony/Component/Form/Extension/Validator/EventListener/ValidationListener.php

```diff
@@ -35,17 +41,27 @@ public static function getSubscribedEvents()
         return [FormEvents::POST_SUBMIT => 'validateForm'];
     }
 
-    public function __construct(ValidatorInterface $validator, ViolationMapperInterface $violationMapper)
+    public function __construct(ValidatorInterface $validator, ViolationMapperInterface $violationMapper, ?EventDispatcherInterface $eventDispatcher = null)
     {
         $this->validator = $validator;
         $this->violationMapper = $violationMapper;
+
+        if (!$this->eventDispatcher) {
```

Comment:
Shouldn't this be the following?
```suggestion
        if (!$eventDispatcher) {
```
`$this->eventDispatcher` is never set at this point.